subliminal/subtitle.py

```diff
@@ -334,17 +349,20 @@ def guess_encoding(self) -> str | None:
 
         return encoding_or_none
 
-    def get_path(self, video: Video, *, single: bool = False) -> str:
+    def get_path(self, video: Video, *, single: bool = False, extension: str | None = None) -> str:
         """Get the subtitle path using the `video`, `language` and `extension`.
 
         :param video: path to the video.
         :type video: :class:`~subliminal.video.Video`
         :param bool single: save a single subtitle, default is to save one subtitle per language.
+        :param (str | None) extension: the subtitle extension, default is to match to the subtitle format.
         :return: path of the subtitle.
         :rtype: str
 
         """
-        return get_subtitle_path(video.name, None if single else self.language)
+        if extension is None:
+            extension = FORMAT_TO_EXTENSION.get(self.subtitle_format, '.txt')
```

Comment:
IMO defaulting to `.srt` would be better here, that's what subliminal has been doing all along

Reply:
Right, it's better to keep the old behavior.
Anyway the fallback should happen only if the `subtitle_format` was not defined and could not be guessed (meaning it's an invalid format).